tests/Utility/ResponseToArrayHelperTest.php

```diff
@@ -42,21 +42,39 @@ protected function setUp(): void
 
     /**
      * Tests the expand parameter compatibility.
+     *
+     * @dataProvider expandCompatibilityDataProvider
      */
-    public function testExpandCompatibility(): void
+    public function testExpandCompatibility($edgeResponse, $hybridResponse): void
     {
         /** @var \Psr\Http\Message\ResponseInterface $response1 */
         $response1 = $this->getMockBuilder(ResponseInterface::class)->getMock();
         $response1->method('getHeaderLine')->willReturn('application/json');
-        $response1->method('getBody')->willReturn('["helloworld","weather"]');
+        $response1->method('getBody')->willReturn($edgeResponse);
         $decodedEdgeStyle = $this->responseToArrayHelper->convertResponseToArray($response1, false);
 
         /** @var \Psr\Http\Message\ResponseInterface $response2 */
         $response2 = $this->getMockBuilder(ResponseInterface::class)->getMock();
         $response2->method('getHeaderLine')->willReturn('application/json');
-        $response2->method('getBody')->willReturn('{"proxies":[{"name":"helloworld"},{"name":"weather"}]}');
+        $response2->method('getBody')->willReturn($hybridResponse);
         $decodedWithCompatibility = $this->responseToArrayHelper->convertResponseToArray($response2, true);
 
         $this->assertEquals($decodedEdgeStyle, $decodedWithCompatibility);
     }
+
+    /**
+     * DataProvider for testExpandCompatibility.
+     *
+     * @return array
+     *   The array of values to run tests on.
+     */
+    public function expandCompatibilityDataProvider() {
```

Comment:
Maybe this should be an array and converted to json via json_encode() so it is simpler to make changes ?

Reply:
Yeah, it would be clearer what the format of Hybrid responses are. Pushed the change. :+1: